libs/accounts/two-factor/src/lib/backup-code.manager.ts

```diff
@@ -0,0 +1,19 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+interface BackupCodeRepository {
+  getBackupCodes(userId: string): string[];
+}
+
+export class BackupCodeManager {
+  constructor(private readonly backupCodeRepository: BackupCodeRepository) {}
+
+  hasBackupCodes(userId: string): { hasBackupCodes: boolean; count: number } {
```

Comment:
Perhaps rename to `getCountForUserId`? This method is implicitly part of the BackupCodeManager class so doesn't need the name of the class within the class.